### PR TITLE
[stable/cluster-autoscaler] add support for configmap for containers environment variables

### LINKED ISSUE #16039: [stable/cluster-autoscaler] allow custom configmaps in container environment variables
**Is your feature request related to a problem? Please describe.**
Allow configmaps to inject metadata for cluster name into the cluster-autoscaler.

### PR DESCRIPTION
**Pull request by ajohnstone**
#### Which issue this PR fixes
  - fixes #16039

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/cluster-autoscaler/templates/deployment.yaml [hunk 1]
```diff
@@ -142,6 +142,13 @@ spec:
             - name: {{ $key }}
               value: "{{ $value }}"
           {{- end }}
+          {{- range $key, $value := .Values.envFromConfigMap }}
```
Comment (yurrriq) [inline stable/cluster-autoscaler/templates/deployment.yaml:145]: I'm not sure I follow. Does the intended usage look like this? What's the point of using a map and ignoring `$key`?

```yaml
envFromConfigMap:
  ignoredKey:
    name: MY_ENV_VAR
    config_name: my-config-map
    config_key: myEnvVar
```

Comment (yurrriq) [reply stable/cluster-autoscaler/templates/deployment.yaml:145]: Maybe we could do something like the following instead.

```yaml
{{- range $key, $value := .Values.envFromConfigMap }}
- name: {{ $key }}
  valueFrom:
    configMapKeyRef:
      name: {{ default (include "cluster-autoscaler.fullname" $) $value.name }}
      key: {{ required "Must specify key!" $value.key }}
{{- end }}
```

```yaml
envFromConfigMap:
  MY_ENV_VAR:
    # name: my-config-map
    key: myEnvVar
```

Comment (yurrriq) [reply stable/cluster-autoscaler/templates/deployment.yaml:145]: Alternatively, I see this pattern used once or twice: https://github.com/helm/charts/blob/4719d54/incubator/common/templates/_envvar.tpl#L9-L19

**Review by yurrriq — CHANGES_REQUESTED** [review]
Thanks for the contribution! I think a few tweaks are in order, and have made some suggestions inline.

File: stable/cluster-autoscaler/templates/deployment.yaml (hunk 1 above)
Comment (ajohnstone) [reply stable/cluster-autoscaler/templates/deployment.yaml:145]: I had this structure, but as environment variable names are unique makes sense to use the form you suggested in your first point. Will amend shortly :)

```
envFromConfigMap:
  - name: MY_ENV_VAR
    config_name: my-config-map
    config_key: myEnvVar
```

E.g.
```
envFromConfigMap:
  MY_ENV_VAR:
    name: my-config-map
    key: myEnvVar
```

Comment (ajohnstone) [reply stable/cluster-autoscaler/templates/deployment.yaml:145]: Updated

**Comment by ajohnstone** [comment]
@yurrriq Please re-review, thanks :)

File: stable/cluster-autoscaler/Chart.yaml [hunk 2]
```diff
@@ -2,7 +2,7 @@ apiVersion: v1
 description: Scales worker nodes within autoscaling groups.
 icon: https://github.com/kubernetes/kubernetes/blob/master/logo/logo.png
 name: cluster-autoscaler
-version: 3.1.0
+version: 3.1.1
```
Comment (yurrriq) [inline stable/cluster-autoscaler/Chart.yaml]: Since this change set adds a feature, let's bump the minor version.
```suggestion
version: 3.2.0
```

**Review by yurrriq — CHANGES_REQUESTED** [review]
I've picked a nit re: version, otherwise this looks good to me. I'll test today.

File: stable/cluster-autoscaler/Chart.yaml (hunk 2 above)
Comment (ajohnstone) [reply stable/cluster-autoscaler/Chart.yaml]: bumped.

**Review by yurrriq — APPROVED** [review]
/lgtm

**Comment by yurrriq** [comment]
Thanks, @ajohnstone!

**Pull request merged**